src/smart-components/group/add-group/users-list.js

```diff
@@ -63,13 +64,31 @@ const createRows = (userLinks) => (data, _expanded, checkedRows = []) => {
     : [];
 };
 
-const UsersList = ({ users, fetchUsers, isLoading, pagination, selectedUsers, setSelectedUsers, userLinks, props }) => {
-  const [filterValue, setFilterValue] = useState('');
-  const [emailValue, setEmailValue] = useState('');
-  const [statusValue, setStatusValue] = useState(['Active']);
+const UsersList = ({ users, fetchUsers, updateUsersFilters, isLoading, pagination, selectedUsers, setSelectedUsers, userLinks, inModal, props }) => {
+  const defaultPagination = useSelector(({ userReducer: { users } }) => ({
+    limit: inModal ? users.meta.limit : users.pagination.limit || defaultSettings.limit,
+    offset: inModal ? users.meta.offset : users.pagination.offset || defaultSettings.offset,
+  }));
+
+  const history = useHistory();
+
+  const stateFilters = useSelector(({ userReducer }) => userReducer.users.filters);
+
+  const [filters, setFilters] = useState({
+    username: (!inModal && stateFilters.username) || '',
+    email: (!inModal && stateFilters.email) || '',
+    status: !inModal && status ? status : ['Active'],
+  });
+
+  useEffect(() => {
+    inModal || setFilters(stateFilters);
+  }, [stateFilters]);
```

Comment:
If `stateFilters` is a nested object, this trigger won't work properly and the effect will be run all the time. This structure will be always evaluated as changed: `{status: ['A']} !== {status: ['A']}`. We will be trigerring the effect even if it's not required. It is safer to break it down to literal values like so:

```jsx
  useEffect(() => {
    inModal || setFilters(stateFilters);
  }, [stateFilters.username, stateFilters.email, stateFilters.status]);
```

Reply:
oh, that makes sense, thank you, I'll fix that